### PR TITLE
feat(sidecar-injector): add useful labels to injected pods

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
# Description

Added two pod labels in addition to `dapr.io/sidecar-injected` (from #5937):
- `dapr.io/app-id`: useful as a target label in exported metrics (for example, to allow for a single generic PodMonitor instead of one per application).
- `dapr.io/metrics-enabled`: useful for label selectors on PodMonitors to avoid scraping pods where metrics are disabled.

## Issue reference

Please reference the issue this PR will close: #4366

> **Note** This issue is closed, but there is a request to reopen.

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by ItalyPaleAle — DISMISSED** [review]

File: pkg/injector/sidecar/consts.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ const (
 	SidecarDebugPortName           = "dapr-debug"
 	SidecarHealthzPath             = "healthz"
 	SidecarInjectedLabel           = "dapr.io/sidecar-injected"
+	SidecarAppIdLabel              = "dapr.io/app-id"
```
Comment (daixiang0) [inline pkg/injector/sidecar/consts.go:30]: Please fix CI

Comment (NickLarsenNZ) [reply pkg/injector/sidecar/consts.go:30]: Oh yes, meant to do that last night.
Shouldn't `make check` pick this up?

I see mention of a specific version of the linter needed in the contributor docs, maybe `make check` needs fixing to pick this up when running locally? Or perhaps I just need to use the devcontainer :)

Comment (NickLarsenNZ) [reply pkg/injector/sidecar/consts.go]: Fixed

Comment (Taction) [reply pkg/injector/sidecar/consts.go]: `make check` should have checked this lint error, and please make sure you are using golangci-lint version v1.51.2.

**Review by berndverst — APPROVED** [review]

**Review by yaron2 — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Comment by yaron2** [comment]
Thanks for this contribution

**Pull request merged**